lib/internal/crypto/webidl.js

```diff
@@ -0,0 +1,705 @@
+'use strict';
+
+// Adapted from the following sources
+// - https://github.com/jsdom/webidl-conversions
+//   Copyright Domenic Denicola. Licensed under BSD-2-Clause License.
+//   Original license at https://github.com/jsdom/webidl-conversions/blob/master/LICENSE.md.
+// - https://github.com/denoland/deno
+//   Copyright Deno authors. Licensed under MIT License.
+//   Original license at https://github.com/denoland/deno/blob/main/LICENSE.md.
+// Changes include using primordials and stripping the code down to only what
+// WebCryptoAPI needs.
+
+const {
+  ArrayBufferIsView,
+  ArrayBufferPrototype,
+  ArrayPrototypePush,
+  ArrayPrototypeSort,
+  DataViewPrototypeGetBuffer,
+  globalThis: {
+    SharedArrayBuffer,
+  },
+  MathPow,
+  MathTrunc,
+  Number,
+  NumberIsFinite,
+  ObjectAssign,
+  ObjectPrototypeIsPrototypeOf,
+  SafeArrayIterator,
+  SafeSet,
+  String,
+  SymbolIterator,
+  TypedArrayPrototypeGetSymbolToStringTag,
+  TypeError,
+  TypedArrayPrototypeGetBuffer,
```

Comment:
nit: `'globalThis' > 'TypedArrayPrototypeGetBuffer'`
```suggestion
  MathPow,
  MathTrunc,
  Number,
  NumberIsFinite,
  ObjectAssign,
  ObjectPrototypeIsPrototypeOf,
  SafeArrayIterator,
  SafeSet,
  String,
  SymbolIterator,
  TypedArrayPrototypeGetSymbolToStringTag,
  TypeError,
  TypedArrayPrototypeGetBuffer,
  globalThis: {
    SharedArrayBuffer,
  },
```